docs/connectors/index.mdx

```diff
@@ -1,11 +1,170 @@
-# Data Sources and Sinks
+# Connectors: connect to data sources and sinks
 
 A Feldera pipeline can process data from multiple heterogeneous sources and
-produce outputs to multiple heterogeneous destinations.  To this end it relies
-on a growing library of input and output connectors.
+produce outputs to multiple heterogeneous destinations. To this end it relies
+on a growing library of **[input](sources)** and **[output](sinks)** connectors.
 
-```mdx-code-block
-import DocCardList from '@theme/DocCardList';
+:::caution
 
-<DocCardList />
+The connectors API is still evolving and might see backwards incompatible changes.
+
+:::
+
+## Basics
+
+The connector object is the abstraction used to define data coming in from data sources
+and going out to data sinks.
+
+* The program of a pipeline defines the tables and views it has,
+  to which input and output connectors can respectively be attached.
+* A connector must explicitly be attached to a table (for input)
+  or view (for output) of a pipeline.
+* A connector can be attached to the tables/views of one or more pipelines,
+  and can even be attached to multiple tables (for input) or views (for output)
+  within the same pipeline.
+* Connectors are managed through the API under the **/v0/connectors/...** endpoints.
+  They can be retrieved
+  [individually](https://www.feldera.com/api/fetch-a-connector-by-name) or
+  [as a list](https://www.feldera.com/api/fetch-connectors-optionally-filtered-by-name-or-id),
+  [created](https://www.feldera.com/api/create-a-new-connector),
+  [fully modified](https://www.feldera.com/api/create-or-replace-a-connector),
+  [partially modified](https://www.feldera.com/api/update-the-name-description-and-or-configuration-of-a-connector) and
+  [deleted](https://www.feldera.com/api/delete-an-existing-connector).
+* If changes are made to the connectors of a running pipeline, they will only be
+  applied when the pipeline is restarted.
+
+> **Note:** these basics apply to all connectors **except** the
+  HTTP [input](/docs/connectors/sources/http) and [output](/docs/connectors/sinks/http) connectors
+  which are not managed by the user, as they directly feed/fetch data into/from a pipeline via
+  dedicated pipeline endpoints.
+
```

Comment:
A connector specification consists of three parts:
* Generic attributes common to all connectors, such as backpressure thresholds
* Transport specification, which defines the data transport to be used by the connector (e.g., Kafka, URL, Delta Lake, etc.) and its configuration.
* Data format specification, which defines the data format for the connector (e.g., JSON, Parquet, or Avro), and its specific dialect 

This architecture allows the user to combine different transports and data formats.

<Links to doc section on input and output transports, and supported data formats>

Reply:
No docs yet though to link to for Avro and Delta Lake.